OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -103,7 +103,7 @@ public override Activity Tick(Actor self)
 				foreach (var n in notify)
 					n.MovingToResources(self, closestHarvestablePosition.Value, this);
 
-				return ActivityUtils.SequenceActivities(mobile.MoveTo(closestHarvestablePosition.Value, 1), new HarvestResource(self), this);
+				return ActivityUtils.SequenceActivities(mobile.MoveTo(closestHarvestablePosition.Value, 2), new HarvestResource(self), this);
```

Comment:
The proper fix will be to reimplement harvesting using child/nested activities.  It would then be easy for the parent activity to check when the target cell has been blocked, and act accordingly.  This is very similar to what `MoveAdjacentTo` already does.